inference.py

```diff
@@ -43,112 +45,223 @@
+    else:
+        padded_arr = np.pad(arr, ((aug, aug), (aug, aug)), mode='reflect')
+
+    return padded_arr
+
+
+def ras2vec(raster_file, output_path):
+    # Create a generic polygon schema for the output vector file
+    i = 0
+    feat_schema = {'geometry': 'Polygon',
+                   'properties': OrderedDict([('value', 'int')])
+                   }
+    class_value_domain = set()
+    out_features = []
+
+    print("   - Processing raster file: {}".format(raster_file))
+    with rasterio.open(raster_file, 'r') as src:
+        raster = src.read(1)
+    mask = raster != 0
+    # Vectorize the polygons
+    polygons = features.shapes(raster, mask, transform=src.transform)
+
+    # Create shapely polygon featyres
+    for polygon in polygons:
+        feature = {'geometry': {
+            'type': 'Polygon',
+            'coordinates': None},
+            'properties': OrderedDict([('value', 0)])}
+
+        feature['geometry']['coordinates'] = polygon[0]['coordinates']
+        value = int(polygon[1])  # Pixel value of the class (layer)
+        class_value_domain.add(value)
+        feature['properties']['value'] = value
+        i += 1
+        out_features.append(feature)
+
+    print("   - Writing output vector file: {}".format(output_path))
+    num_layers = list(class_value_domain)  # Number of unique pixel value
+    for num_layer in num_layers:
+        polygons = [feature for feature in out_features if feature['properties']['value'] == num_layer]
+        layer_name = 'vector_' + str(num_layer).rjust(3, '0')
+        print("   - Writing layer: {}".format(layer_name))
+
+        with fiona.open(output_path, 'w',
+                        crs=to_string(src.crs),
+                        layer=layer_name,
+                        schema=feat_schema,
+                        driver='GPKG') as dest:
+            for polygon in polygons:
+                dest.write(polygon)
+    print("")
+    print("Number of features written: {}".format(i))
+
+
+def gen_img_samples(src, chunk_size, *band_order):
+    subdiv = 2.0
+    step = int(chunk_size / subdiv)
+    for row in range(0, src.height, step):
+        for column in range(0, src.width, step):
+            window = Window.from_slices(slice(row, row + chunk_size),
+                                        slice(column, column + chunk_size))
+            if band_order:
+                window_array = reshape_as_image(src.read(band_order[0], window=window))
+            else:
+                window_array = reshape_as_image(src.read(window=window))
+            if window_array.shape[0] < chunk_size or window_array.shape[1] < chunk_size:
+                window_array = _pad_diff(window_array, window_array.shape[0], window_array.shape[1], chunk_size)
+            window_array = _pad(window_array, chunk_size)
+
+            yield window_array, row, column
 
 
 @torch.no_grad()
-def segmentation(img_array,
+def segmentation(param,
```

Comment:
here again, docstrings could help understand what arguments are expected. Not sure I understand the tp_mem argument. Does it stand for "temporary memory"? It's the image to be inferred?

Reply:
I would add docstrings amd make proper comments. tp_mem is a temp_file to write very large arrays to disk at inference